src/linear_algebra/quadratic_form.lean

```diff
@@ -805,3 +806,151 @@ theorem exists_orthogonal_basis [hK : invertible (2 : K)]
 let ⟨v, hv₁, _⟩ := exists_orthogonal_basis' hB₁ hB₂ in ⟨v, hv₁⟩
 
 end bilin_form
+
+-- TODO: Remove both instances after #7438 is merged
+instance units.distrib_mul_action : distrib_mul_action (units R) R :=
+{ smul := λ r s, (r : R) * s,
+  one_smul := one_mul,
+  mul_smul := λ _ _, mul_assoc _ _,
+  smul_add := λ _ _, mul_add _ _,
+  smul_zero := λ _, mul_zero _ }
+
+instance units.smul_comm_class : smul_comm_class (units R₁) R₁ R₁ :=
+{ smul_comm := λ _, mul_left_comm _ }
+
+namespace quadratic_form
+
+open_locale big_operators
+
+open finset bilin_form
+
+variables {M₁ : Type*} [add_comm_group M₁] [module R M₁]
+variables {ι : Type*} [fintype ι] {v : basis ι R M}
```

Comment:
@Vierkantor, do you think this should be using `ι →₀ R` instead of `fintype`?

Reply:
Ah, looking at this again this makes things awkward, as `quadratic_form.proj` doesn't work on finsupp.